### PR TITLE
feat(apple): Add docs on file I/O APM

### PR DESCRIPTION
**Pull request by brustolin**
Documentation for file I/O auto transactions experimental feature added by https://github.com/getsentry/sentry-cocoa/pull/1557

### CODE REVIEW

File: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 1]
```diff
@@ -86,6 +86,38 @@ SentrySDK.start { options in
 [NSURLSession]: https://developer.apple.com/documentation/foundation/nsurlsession
 [NSURLConnection]: https://developer.apple.com/documentation/foundation/nsurlconnection
 
+## File I/O Instrumentation
+
+<Alert level="info" title="Important">
+
+This is an experimental feature. Use it with caution.
```
Comment (bruno-garcia) [inline src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx:93]:
```suggestion
This is an experimental feature and requires an opt-in.
```

File: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 2]
```diff
@@ -86,6 +86,38 @@ SentrySDK.start { options in
...
 
+## File I/O Instrumentation
+
+<Alert level="info" title="Important">
+
+This is an experimental feature and requires an opt-in.
+
+</Alert>
+
+The Sentry SDK adds spans for file I/O operations to ongoing transactions bound to the Scope. Currently, the SDK supports operations with [NSData][NSData], but many other Apple APIs like [NSFileManager][NSFileManager] uses [NSData][NSData].
```
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx:97]:
```suggestion
The Sentry SDK adds spans for file I/O operations to ongoing transactions bound to the Scope. Currently, the SDK supports operations with [NSData][NSData], but many other APIs like [NSFileManager][NSFileManager] use [NSData][NSData].
```

Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx:97]: `m`: Can we list the Apple APIs we know that use NSData internally, please?

File: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 3]
```diff
@@ -86,6 +86,38 @@ SentrySDK.start { options in
...
+This feature is experimental and is disable by default. 
+
+To enable File I/O instrumentation:
```
Comment (philipphofmann) [inline src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx:100]:
```suggestion
To enable file I/O instrumentation:
```

**Review by philipphofmann — COMMENTED** [review]
A few typos and suggestions. It's worth noting we should only merge this once 7.9.0 is released.

File: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx (hunk 2 above)
Comment (brustolin) [reply src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx]: Well, there is a ton of APIs that use NSData internally. 
But if you mean APIs available to load data directly I could think of NSFileManager, NSString and NSBundle.
I would add this as examples, but I don't believe a complete list is relevant here.

File: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx [hunk 4]
```diff
@@ -86,6 +86,38 @@ SentrySDK.start { options in
...
 
+## File I/O Instrumentation
+
+<Alert level="info" title="Important">
+
+This is an experimental feature and requires an opt-in. Experimental features are still a work-in-progress and may have bugs. We recognize the irony.
+
+</Alert>
+
+The Sentry SDK adds spans for file I/O operations to ongoing transactions bound to the Scope. Currently, the SDK supports operations with [NSData][NSData], but many other APIs like [NSFileManager][NSFileManager] use [NSData][NSData].
```
Comment (imatwawana) [inline src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx:97]: I know it's not consistent across the board, but mostly we don't capitalize "Scope" in this context in the docs. 

```suggestion
The Sentry SDK adds spans for file I/O operations to ongoing transactions bound to the scope. Currently, the SDK supports operations with [NSData][NSData], but many other APIs like [NSFileManager][NSFileManager] use [NSData][NSData].
```

**Review by imatwawana — APPROVED** [review]
One small wording fix

**Review by philipphofmann — COMMENTED** [review]
I completely forgot that we also need to update the swizzling page here https://docs.sentry.io/platforms/apple/guides/ios/configuration/swizzling/. Can you please do that @brustolin?

**Comment by brustolin** [comment]
> I completely forgot that we also need to update the swizzling page here https://docs.sentry.io/platforms/apple/guides/ios/configuration/swizzling/. Can you please do that @brustolin?

Sure, I will update it.

**Review by bruno-garcia — APPROVED** [review]
:shipit:

**Review by philipphofmann — APPROVED** [review]
LGTM 🚀

**Pull request merged**